### DISCUSSION TITLE
Combine Tauri generated android code with my native android code

### DISCUSSION THREAD

**Discussion by umesh-umesh** [tauri-apps/tauri#11444] [Q&A]
I have a chat android app that is completely built on only single activity with multiple fragments. So now can i add another ecommerce tauri generated android code to my native android chat app code? 

Or is there a way i can import tauri generated android code main activity as a fragment in my native code?

Previously i had same issue with capacitorjs also, but was able to add capacitorjs generated android code main activity as a fragment to my native  code. But i am unable to figure out same in tauri generated android code. Is it possible to use main activity of tauri generated android code as fragment to my native code main activity?

**Comment by blafed** [comment]
Yeah, I want to do something similar but don't know how.

I want to add android Canvas component as background under the WebView.
so my app looks like this:
- android canvas renderered as background
- then the tauri webview

**Comment by ctsstc** [comment]
I was interested in this to see if it would be possible to use the Android "Native Sign In" as mentioned here:
<https://supabase.com/docs/guides/auth/social-login/auth-google?queryGroups=platform&platform=android&queryGroups=environment&environment=client&queryGroups=framework&framework=sveltekit#using-google-sign-in-on-android>

**Comment by unlocomqx** [comment]
can probably be done using a custom Tauri plugin

https://tauritutorials.com/blog/develop-a-tauri-plugin-for-android

https://v1.tauri.app/v1/guides/features/plugin/

**Comment by Ggcpp** [comment]
Any news about this? did someone figure it out?
